Sources/AsyncDNSResolver/dnssd/DNSResolver_dnssd.swift

```diff
@@ -138,32 +123,54 @@ struct DNSSD {
                 handler.handleRecord(errorCode: errorCode, data: rdata, length: rdlen)
             }
 
-            let serviceRefPtr = UnsafeMutablePointer<DNSServiceRef?>.allocate(capacity: 1)
-            defer { serviceRefPtr.deallocate() }
+            let serviceRefPointer = UnsafeMutablePointer<DNSServiceRef?>.allocate(capacity: 1)
+
+            // Wrap 'handler' into a pointer so we can pass it to DNSServiceQueryRecord
+            let replyHandlerPointer = UnsafeMutablePointer<QueryReplyHandler>.allocate(capacity: 1)
+            replyHandlerPointer.initialize(repeating: handler, count: 1)
 
             // Run the query
             let _code = DNSServiceQueryRecord(
-                serviceRefPtr,
+                serviceRefPointer,
                 kDNSServiceFlagsTimeout,
                 0,
                 name,
                 UInt16(type.kDNSServiceType),
                 UInt16(kDNSServiceClass_IN),
                 callback,
-                handlerPointer
+                replyHandlerPointer
             )
 
             // Check if query completed successfully
             guard _code == kDNSServiceErr_NoError else {
                 return continuation.finish(throwing: AsyncDNSResolver.Error(dnssdCode: _code))
             }
 
-            // Read reply from the socket (blocking) then call reply handler
-            DNSServiceProcessResult(serviceRefPtr.pointee)
-            DNSServiceRefDeallocate(serviceRefPtr.pointee)
+            let serviceSockFD = DNSServiceRefSockFD(serviceRefPointer.pointee)
+            guard serviceSockFD != -1 else {
+                return continuation.finish(throwing: AsyncDNSResolver.Error(code: .internalError, message: "Failed to access the DNSSD service socket"))
```

Comment:
Aren't we leaking the `serviceRefPointer` and `replyHandlerPointer` in this branch?